replay/obp_evaluation/replay_offline.py

```diff
@@ -0,0 +1,237 @@
+
+        :param timestamp: Moment of time when user interacted with corresponding item.
+
+        :param context: Context vectors observed for each data, i.e., :math:`x_i`.
+
+        :param action_context: Context vectors observed for each action.
+        """
+
+        log = convert2spark(obp2df(action, reward, timestamp))
+        self.log = log
+
+        user_features = None
+        self.max_usr_id = reward.shape[0]
+
+        if context is not None:
+            user_features = convert2spark(context2df(context,
+                                                     np.arange(context.shape[0]),
+                                                     "user"))
+
+        if action_context is not None:
+            self.item_features = convert2spark(context2df(action_context,
+                                                          np.arange(self.n_actions),
+                                                          "item"))
+
+        self.replay_model._fit_wrap(log, user_features, self.item_features)
+
+    # pylint: disable=arguments-renamed
+    def predict(self,
+                n_rounds: int = 1,
+                context: np.ndarray = None) -> np.ndarray:
+        '''Predict best actions for new data.
+        Action set predicted by this `predict` method can contain duplicate items.
+        If a non-repetitive action set is needed, please use the `sample_action` method.
+
+        :context: Context vectors for new data.
+
+        :return: Action choices made by a classifier, which can contain duplicate items.
+            If a non-repetitive action set is needed, please use the `sample_action` method.
+        '''
+
+        user_features = None
+        if context is not None:
+            user_features = convert2spark(context2df(context,
+                                                     np.arange(self.max_usr_id,
+                                                               self.max_usr_id + n_rounds),
+                                                     'user'))
+
+        users = convert2spark(pd.DataFrame({"user_idx": np.arange(self.max_usr_id,
+                                                                  self.max_usr_id + n_rounds)}))
+        items = convert2spark(pd.DataFrame({"item_idx": np.arange(self.n_actions)}))
+
+        self.max_usr_id += n_rounds
+
+        action_dist = self.replay_model._predict_proba(self.log,
+                                                       self.len_list,
+                                                       users,
+                                                       items,
+                                                       user_features=user_features,
+                                                       item_features=self.item_features,
+                                                       filter_seen_items=False)
+
+        assert np.allclose(action_dist.sum(1), np.ones(shape=(n_rounds, self.len_list)))
```

Comment:
Consider moving this line to `tests/obp_evaluation/test_obp_learner.py::test_predict`